src/java/htsjdk/samtools/util/Locatable.java

```diff
@@ -22,4 +22,63 @@
      * @return 1-based closed-ended position, undefined if getContig() == null
      */
     int getEnd();
+
+
+    /**
+    * @return the 0-based start position (from the GA4GH spec).
+    */
+    default long getGA4GHStart() {return this.getStart() - 1; }
+
+    /**
+    * @return the typical end spans are [zero-start,end) (from the GA4GH spec).
+    */
+    default long getGA4GHEnd() { return this.getEnd(); }
+
+    /**
+     * @return number of bases covered by this interval (will always be > 0)
+     */
+    default int size() {
+        return getEnd() - getStart() + 1;
+    }
+
+    /**
+     * Determines whether this interval overlaps the provided locatable.
+     *
+     * @param other interval to check
+     * @return true if this interval overlaps other, otherwise false
+     */
+    default boolean overlaps(Locatable other) {
+        return overlapsWithMargin(other, 0);
+    }
+
+    /**
+      * Determines whether this interval comes within "margin" of overlapping the provided locatable.
+      * This is the same as plain overlaps if margin=0.
+      *
+      * @param other interval to check
+      * @param margin how many bases may be between the two interval for us to still consider them overlapping.
+      * @return true if this interval overlaps other, otherwise false
+      */
+     default boolean overlapsWithMargin(Locatable other, int margin) {
```

Comment:
This is a little confusing - partially because of the naming.  How about renaming this to something like `withinDistanceof(Locatable other, int distance)`?